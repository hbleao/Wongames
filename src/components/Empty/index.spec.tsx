import React from 'react';
import { screen } from '@testing-library/react';

import { Empty } from '.';

import { renderWithTheme } from 'utils/tests/renderWithTheme';

type MakeSutProps = {
  title?: string;
  description?: string;
  hasLink?: boolean;
};

const makeSut = ({
  title = 'a simple title',
  description = 'a simple description',
  hasLink
}: MakeSutProps) => {
  const sut = renderWithTheme(
    <Empty title={title} description={description} hasLink={hasLink} />
  );

  return {
    sut
  };
};

describe('Empty', () => {
  it('should render correctly', () => {
    const { sut } = makeSut({ hasLink: true });

    const emptyImage = screen.getByRole('img', {
      name: /a gamer in a couch playing video game/i
    });

    expect(emptyImage).toBeInTheDocument();

    const heading = screen.getByRole('heading', { name: /a simple title/i });
    expect(heading).toBeInTheDocument();

    const description = screen.getByText(/a simple description/i);
    expect(description).toBeInTheDocument();

    const link = screen.getByRole('link', { name: /go back to store/i });
    expect(link).toHaveAttribute('href', '/');

    expect(sut).toMatchSnapshot();
  });

  it('should not render link when hasLink is not passed', () => {
    makeSut({});

    const link = screen.queryByRole('link', { name: /go back to store/i });
    expect(link).not.toBeInTheDocument();
  });
});
