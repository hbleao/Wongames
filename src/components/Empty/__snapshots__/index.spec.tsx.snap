// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c4 {
  color: #FAFAFA;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

<body>
    <div>
      <main
        class="c0"
      >
        <img
          alt="a gamer in a couch playing video game"
          class="c1"
          role="img"
          src="/img/empty.svg"
        />
        <h2
          class="c2"
        >
          a simple title
        </h2>
        <p
          class="c3"
        >
          a simple description
        </p>
        <a
          class="c4"
          href="/"
        >
          <span>
            Go back to store
          </span>
        </a>
      </main>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c4 {
  color: #FAFAFA;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  background-image: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

<div>
    <main
      class="c0"
    >
      <img
        alt="a gamer in a couch playing video game"
        class="c1"
        role="img"
        src="/img/empty.svg"
      />
      <h2
        class="c2"
      >
        a simple title
      </h2>
      <p
        class="c3"
      >
        a simple description
      </p>
      <a
        class="c4"
        href="/"
      >
        <span>
          Go back to store
        </span>
      </a>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
